contracts/utils/Strings.sol

```diff
@@ -426,6 +434,43 @@ library Strings {
         return value;
     }
 
+    /**
+     * @dev Escape special characters in JSON strings. This can be useful to prevent JSON injection in NFT metadata.
+     *
+     * WARNING: This function should only be used in double quoted JSON strings. Single quotes are not escaped.
+     */
+    function escapeJSON(string memory input) internal pure returns (string memory) {
+        bytes memory buffer = bytes(input);
+        bytes memory output = new bytes(2 * buffer.length); // worst case scenario
```

Comment:
I don't like that this is allocating worst-case scenario memory. Ideally, it should be just `buffer.length`, iterate and count the characters to escape, then indeed escape them. The tradeoff here is perhaps a bit more computation (i.e. more checks over) so I'm not convinced it's an actual saving.

Reply:
We tested to have two loops:

- one loop that counts the times escaping is needed
- then we allocate exactly the right size
- then we loop a second time.

It ends up being more expensive.

---

An alternative is to do all the allocation manually (basically we move the fmp ourselves)